sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -102,31 +106,31 @@ public Mono<Void> authorizeRequest(HttpPipelineCallContext context) {
 
             // If this policy doesn't have challenge parameters cached try to get it from the static challenge cache.
             if (this.challenge == null) {
-                String authority = getRequestAuthority(request);
-                this.challenge = CHALLENGE_CACHE.get(authority);
+                this.challenge = CHALLENGE_CACHE.get(getRequestAuthority(request));
             }
 
             if (this.challenge != null) {
                 // We fetched the challenge from the cache, but we have not initialized the scopes in the base yet.
                 TokenRequestContext tokenRequestContext = new TokenRequestContext()
                     .addScopes(this.challenge.getScopes())
-                    .setTenantId(this.challenge.getTenantId());
+                    .setTenantId(this.challenge.getTenantId())
+                    .setClaims(this.challenge.getClaims());
```

Comment:
Question, shouldn't the Claims be only added if CAE is enabled? If so, should we also introduce a flag for these classes?

Also, I think we have to tell the server that CAE is enabled by adding `.setCaeEnabled(true)`

Reply:
You're right @JonathanCrd, we should enable CAE in the `TokenRequestContext`. I also made sure to only set the claims when a claims challenge reaches the client.